python/ray/_private/job_manager/job_manager.py

```diff
@@ -152,21 +172,39 @@ def run(self, cmd: str):
             stdout_path, stderr_path = self._log_client.get_log_file_paths(
                 self._job_id)
 
-            exit_code = exec_cmd_logs_to_file(cmd, stdout_path, stderr_path)
-        finally:
-            # 3) Once command finishes, update status to SUCCEEDED or FAILED.
-            if exit_code == 0:
-                self._status = JobStatus.SUCCEEDED
+            task, child_proc = await self._exec_cmd(cmd, stdout_path,
+                                                    stderr_path)
+
+            finished, _ = await asyncio.wait(
+                [task, self._stop_event.wait()], return_when=FIRST_COMPLETED)
+
+            if self._stop_event.is_set():
+                self._status_client.put_status(self._job_id, JobStatus.STOPPED)
+                task.cancel()
+                child_proc.kill()
```

Comment:
I think the ordering here is backwards -- we should kill the child process before marking the status as `STOPPED`. I don't think this necessarily matters right now but could introduce a race condition/inconsistency issue.

Reply:
good call, i previously forgot to put task in asyncio.task and tests will only pass if we set status first. But with correct changes we can put the status change after coro + child_proc killed still with passing tests.